src/qcodes/extensions/_refactor.py

```diff
@@ -53,69 +52,35 @@ def visit_Arg(self, node: cst.Arg) -> None:
         if self._call_stack[-1] != "self.add_parameter":
             return
 
-        first_positional_arg_is_str = self._arg_num[
-            self._call_stack[-1]
-        ] == 1 and m.matches(
-            node,
-            matcher=m.Arg(value=m.SimpleString(), keyword=None),
-        )
-        arg_is_name = m.matches(
-            node, m.Arg(keyword=m.Name(value="name"), value=m.SimpleString())
-        )
-
-        arg_is_docstring = m.matches(
-            node,
-            matcher=m.Arg(
-                keyword=m.Name(value="docstring"),
-                value=m.OneOf(m.SimpleString(), m.ConcatenatedString()),
-            ),
-        )
-
-        arg_is_docstring_dedent = m.matches(
-            node,
-            m.Arg(
-                keyword=m.Name(value="docstring"),
-                value=m.Call(args=[m.Arg(m.SimpleString())]),
-            ),
-        )
-
-        arg_is_parameter_class = m.matches(
-            node,
-            matcher=m.Arg(keyword=m.Name(value="parameter_class"), value=m.Name()),
-        )
-
-        second_positional_arg_is_str = self._arg_num[
-            self._call_stack[-1]
-        ] == 2 and m.matches(
-            node,
-            matcher=m.Arg(value=m.Name(), keyword=None),
-        )
-
-        if first_positional_arg_is_str or arg_is_name:
-            self.annotations.name = cst.ensure_type(
-                node.value, cst.SimpleString
-            ).raw_value
-
-        if arg_is_docstring:
-            self.annotations.docstring = str(
-                cast(
-                    Union[cst.SimpleString, cst.ConcatenatedString], node.value
-                ).evaluated_value
-            )
-        if arg_is_docstring_dedent:
-            self.annotations.docstring = dedent(
-                str(
-                    cst.ensure_type(
-                        cst.ensure_type(node.value, cst.Call).args[0].value,
-                        cst.SimpleString,
-                    ).evaluated_value
-                )
-            ).strip()
-
-        if arg_is_parameter_class or second_positional_arg_is_str:
-            self.annotations.parameter_class = cst.ensure_type(
-                node.value, cst.Name
-            ).value
+        match node:
+            case cst.Arg(
+                value=cst.SimpleString(e_value), keyword=None
+            ) if self._arg_num[self._call_stack[-1]]:
+                self.annotations.name = e_value.strip("\"'")
+            case cst.Arg(
+                keyword=cst.Name(value="name"), value=cst.SimpleString(e_value)
+            ):
+                self.annotations.name = e_value.strip("\"'")
+            case cst.Arg(
```

Comment:
this looks indeed more readable except the variables like "arg_is_docstring_dedent" were really helping in the previous version, and now they are not there :( could they be added to the case statements? or perhaps at least as comments?

Reply:
The case statement is more than a boolean so it cannot be extracted into a variable but we can at least add comments